### PR TITLE
[None][feat] Use XQA JIT impl by default and mitigate perf loss with sliding window

### PR DESCRIPTION
**Pull request by pengbowang-nv**
## Summary by CodeRabbit

* **New Features**
  * Expanded token-per-page configuration options to support 32 tokens in addition to existing values.

* **Improvements**
  * Enhanced sliding window attention masking logic for improved accuracy in specific inference scenarios.
  * Streamlined kernel implementation selection for more consistent performance behavior.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

## Description

This PR uses XQA JIT for all cases.

Current XQA has 2 execution path, one is to use legacy pre-compiled cubin and the other is JIT.
With most kernel has been moved to JIT after [https://github.com/NVIDIA/TensorRT-LLM/pull/6078](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FNVIDIA%2FTensorRT-LLM%2Fpull%2F6078&data=05%7C02%7Csephw%40nvidia.com%7Ccdafdb9ba1b84de05e8508de4f918cc4%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C639035683904295966%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=utpybEA4RUFJ18WnGV3gKgiwccVoiSAgzb9rKpMuI%2B4%3D&reserved=0) , remaining kernel are only speculative kernel withfp/bf16 kvcache and 64 % (num_q_head / num_kv_heads) != 0. Some affected models include: Mixtral-8x22B-Instruct-v0.1, Qwen2.5-VL-7B-Instruct, Llama-3.2-3B-Instruct, ...

If fully converted to JIT, we can avoid maintaining 2 separate host invoke code, and remove hundreds of cubins.

This PR remove pre-compiled cubin path from normal execution but keep the code and knob for a release. Then we remove pre-compiled cubin completely after 1-2 releases.

Also, this PR tried to mitigate a perf issue of sliding window with speculative after #8383 . Before this PR, enabling sliding window but not using it at runtime (pass max window size with a size larger than KV lens) will suffer a 5% perf loss compared to disabling sliding window feature completely. Below are some perf data between pre-compiled version, jit version, and jit version with the fix.

Using Qwen3-8B, spec length = 3, ISL=4096, OSL=1024

|Time (relative, lower is better)|Pre-compile(Legacy Cubin)|JIT (Before mitigation)|JIT|JIT Without Sliding Window|
|-|-|-|-|-|
|SM90|1|0.912|0.983|0.981|
|SM89|1|0.986|0.991|0.991|

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Review by jhaotingc — APPROVED** [review]
LGTM! Thanks for resolving perf issue when SLIDING_WINDOW=1!
